packages/vx-tooltip/Readme.md

```diff
@@ -4,10 +4,127 @@
+| className | string           | --      | Adds a class (in addition to `vx-tooltip-portal`) to the tooltip container    |
+| style     | object           | --      | Sets / overrides any styles on the tooltip container (including top and left) |
+| children  | node             | --      | Sets the children of the tooltip, i.e., the actual content                    |
+
+#### TooltipWithBounds
+
+This tooltip component is exactly the same as `Tooltip` above, but it is aware of its boundaries meaning that it will flip left/right and bottom/top based on whether it would overflow its parent's boundaries. It accepts the following props, and will spread any additional props on the Tooltip component (i.e., ...restProps):
+
+| Name        | Type   | Default | Description                                                                                                                                                                      |
+| :---------- | :----- | :------ | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| left        | number | --      | The horizontal position of the cursor, tooltip will be place to the left or right of this coordinate depending on the width of the tooltip and the size of the parent container. |
+| top         | number | --      | The vertical position of the cursor, tooltip will be place to the bottom or top of this coordinate depending on the height of the tooltip and the size of the parent container.  |
+| offsetLeft  | number | 10      | Horizontal offset of the tooltip from the passed `left` value, functions as a horizontal padding.                                                                                |
+| offsetRight | number | 10      | Vertical offset of the tooltip from the passed `top` value, functions as a vertical padding.                                                                                     |
+| style       | object | --      | Sets / overrides any styles on the tooltip container (including top and left)                                                                                                    |
+| children    | node   | --      | Sets the children of the tooltip, i.e., the actual content                                                                                                                       |
+
+Note that this component is positioned using a `transform`, so overriding `left` and `top` via styles may have no effect.
+
+### Examples
+#### useTooltip For Functional Components
+
+```js
+import { useTooltip, TooltipWithBounds } from '@vx/tooltip';
+import { localPoint } from '@vx/event';
+
+const ChartWithTooltip = () => {
+  const {
+    tooltipData,
+    tooltipLeft,
+    tooltipTop,
+    tooltipOpen,
+    showTooltip,
+    hideTooltip,
+  } = useTooltip();
```

Comment:
Is it possible to create a “Tooltip” class and return a `tooltip` instance with the hook? `const [tooltip, updateTooltip] = useTooltip()` feels a more consistent to how people would use other hooks. It’s also more scalable in case we want to add more API to the tooltip.

Reply:
I think returning  an `object` of the same `prop`s that are passed via `withTooltip` makes sense. 

I've seen several examples of hooks that use this pattern (e.g., `react-three-fiber`'s [`useThree` hook](https://github.com/react-spring/react-three-fiber#usethree-sharedcanvascontext)), and it would avoid a breaking change for `withTooltip`.